src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslCertificateAssetDownloader.cs

```diff
@@ -91,27 +93,42 @@ internal static class OpenSslCertificateAssetDownloader
                 // We're not going to report this error to a user, so clear it
                 // (to avoid tainting future exceptions)
                 Interop.Crypto.ErrClearError();
+                OpenSslX509ChainEventSource.Log.InvalidDownloadedOcsp();
             }
 
             return resp;
         }
 
         private static byte[]? DownloadAsset(string uri, TimeSpan downloadTimeout)
         {
-            if (s_downloadBytes != null && downloadTimeout > TimeSpan.Zero)
+            if (s_downloadBytes is null)
             {
-                long totalMillis = (long)downloadTimeout.TotalMilliseconds;
-                CancellationTokenSource? cts = totalMillis > int.MaxValue ? null : new CancellationTokenSource((int)totalMillis);
+                OpenSslX509ChainEventSource.Log.HttpClientNotAvailable();
+                return null;
+            }
 
-                try
-                {
-                    return s_downloadBytes(uri, cts?.Token ?? default);
-                }
-                catch { }
-                finally
-                {
-                    cts?.Dispose();
-                }
+            if (downloadTimeout <= TimeSpan.Zero)
+            {
+                OpenSslX509ChainEventSource.Log.DownloadTimeExceeded();
+                return null;
+            }
+
+            long totalMillis = (long)downloadTimeout.TotalMilliseconds;
+            OpenSslX509ChainEventSource.Log.AssetDownloadStart(totalMillis, uri);
+
+            CancellationTokenSource? cts = totalMillis > int.MaxValue ? null : new CancellationTokenSource((int)totalMillis);
+            byte[]? ret = null;
+
+            try
+            {
+                ret = s_downloadBytes(uri, cts?.Token ?? default);
+                return ret;
+            }
+            catch { }
```

Comment:
This empty catch looks... interesting... from a tracing perspective.

Reply:
I thought about it, but it didn't feel like it would be all that helpful... anything that shows up as a 0 byte response means you'd want to also be processing traces from the Http EventSource (protocol error, network error, task timed out, etc).